test/progress.ts

```diff
@@ -14,35 +14,37 @@ describe('Migrator Progress Reporting', () => {
     };
 
     const migrator = new Migrator(config, log);
+    const progressResults: string[] = [];
 
     migrator.add({
       id: '1',
-      up: function(cb) {
+      up: async function() {
         this.log('1');
-        cb();
       }
     });
 
     migrator.add({
       id: '2',
-      up: function(cb) {
+      up: async function() {
         this.log('2');
-        cb();
       }
     });
 
-    let resultsCount = 0;
-
-    migrator.migrate((err) => {
-      if (err) return done(err);
-    }, (migrationId, migrationRes) => {
-      const status = migrationRes?.status;
-      if (status !== 'ok') {
-        const message = `Error running ${migrationId}, result is ${status}`;
-        return done(new Error(message));
-      }
-      resultsCount += 1;
-      if (resultsCount === 2) done();
+    await new Promise<void>((resolve, reject) => {
+      let resultsCount = 0;
+      migrator.migrate((migrationId, migrationRes) => {
+        const status = migrationRes?.status;
+        if (status !== 'ok') {
+          reject(new Error(`Error running ${migrationId}, result is ${status}`));
+          return;
+        }
+        progressResults.push(migrationId);
+        resultsCount += 1;
+        if (resultsCount === 2) resolve();
+      }).catch(reject);
     });
```

Comment:
```suggestion]
      let resultsCount = 0;
      await migrator.migrate((migrationId, migrationRes) => {
        const status = migrationRes?.status;
        if (status !== 'ok') {
          reject(new Error(`Error running ${migrationId}, result is ${status}`));
          return;
        }
        progressResults.push(migrationId);
        resultsCount += 1;
        if (resultsCount === 2) resolve();
      }).catch(reject);
```

Am I crazy, or is this `new Promise<void>` thing totally unnecessary?

Reply:
Yup, unnecessary. Removed in the next PR (vitest) I think.